packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -615,55 +481,47 @@ const DataGrid = injectIntl((props) => {
   // -------------------------------------
 
   return (
-    <div ref={gridContainerRef} className={cx('data-grid-container')}>
-      <table
-        ref={gridRef}
-        id={id}
-        role="grid"
-        aria-labelledby={ariaLabelledBy}
-        aria-label={ariaLabel}
-        className={cx('data-grid', theme.className)}
-        onKeyDown={handleKeyDown}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        onMouseDown={onMouseDown}
-        tabIndex={0}
-        {...(activeIndex != null && { onMouseUp, onMouseMove, onMouseLeave: onMouseUp })}
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+      tabIndex={0}
+      ref={gridContainerRef}
+      onKeyDown={handleKeyDown}
+      onMouseDown={onMouseDown}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      id={id}
+      className={cx('data-grid-container')}
+    >
+      <GridContext.Provider value={{
+        role: GridConstants.GRID,
+        setCellAriaLiveMessage,
+        tableRef: grid,
+        tableContainerRef,
+      }}
```

Comment:
You did not memoize your provider value.  This will cause many unnecessary rerenders.